Allow splitting translations (#52)
grammyjs/i18n: README.md

## Example

An example project structure, you can also seperate your translations into different files to make it easier to maintain large projects.

> Nested translations don't change how you use translation keys, so nothing should break if you decide to use them.

Reviewer: Can this be a bit more clear?
The file location or name is irrelevant? Only the content of the files is relevant?

Author: Not sure if I fully understand your comment, but would something like this clear things up?

```suggestion
By splitting translations you don't change how you retrieve the keys contained within them, so for example, a key called `greeting` which is located in both the `locales/it.ftl` and `locales/en/dialogues/greeting.ftl` files can be retrieved by simply using `ctx.t("greeting")`.
```

Reviewer 2: Yes, I like this suggestion

Reviewer: I would switch to „Located in either …“. Then the key is not duplicated somewhere which might create another confusion. Otherwise I like the new phrasing a lot better as it is way more clear what to expect.

Author: Good point, I've fixed this in the latest commit, feel free to resolve if you think everything is clear enough.

## Example

An example project structure, you can also seperate your translations into different files to make it easier to maintain large projects.

> Nested translations don't change how you use translation keys, so nothing should break if you decide to use them.

Reviewer: ```suggestion
Nested translations don't change how you use translation keys, so nothing breaks when you decide to use them.
```

Reviewer: Maybe also a different phrasing? (Thinking out loud)
```suggestion
Nested translations don't change how you use translation keys, so they stay them regardless of their containing file.
You can move them without change between files.
```

Author: I was planning to remove this line since (to me at least) it seems too similar to the one on line 38.
Perhaps we can just replace the whole paragraph with something like this:

"An example featuring both nested (`locales/en/...`) and standard (`locales/it.ftl`) file structure variants. Nested translations allow you to seperate your keys into different files, making it easier to maintain larger projects while also letting you use the standard variant at the same time. Using a nested file structure alongside the standard variant won't break any existing translations."

Reviewer 2: @EdJoPaTo can this be resolved? Do you want to do a proper review, too?

Reviewer: i'd like to but due to not much time currently I prefer to skip it for now. I have more time in two weeks but I would hope its resolved until then 😇

Reviewer 2: Alright. 

@strbit your suggestion works for me. Also, the more important docs are found on the website anyway, so I wouldn't want to nitpick this. Feel free to change it and then tag me or request a review when you're ready to have this merged.

Author: I’ll resolve this as I've implemented this suggestion in commit 08b431f 👌

```
.
├── locales/
│   ├── en/
│   │   ├── dialogues/
│   │   │   ├── greeting.ftl
│   │   │   └── goodbye.ftl
│   │   └── help.ftl
│   ├── it.ftl
│   └── ru.ftl
└── bot.ts
```

> By splitting translations you don't change how you retrieve the keys contained within them, so for example, a key called `greeting` which is located in either `locales/it.ftl` or `locales/en/dialogues/greeting.ftl` can be retrieved by simply using `ctx.t("greeting")`.

Reviewer: Oh, it should probably still be in the same language 😇
```suggestion
By splitting translations you don't change how you retrieve the keys contained within them, so for example, a key called `greeting` which is located in either `locales/en.ftl` or `locales/en/dialogues/greeting.ftl` can be retrieved by simply using `ctx.t("greeting")`.
```

└── bot.ts
```

> By splitting translations you don't change how you retrieve the keys contained within them, so for example, a key called `greeting` which is located in either the `locales/it.ftl` and `locales/en/dialogues/greeting.ftl` files can be retrieved by simply using `ctx.t("greeting")`.

Reviewer: I think the sentence can be simplified a bit more.
```suggestion
By splitting translations you don't change how you retrieve the keys contained within them, so for example, a key called `greeting` which is located in either `locales/it.ftl` or `locales/en/dialogues/greeting.ftl` can be retrieved by simply using `ctx.t("greeting")`.
```
